source/core/sudoku/grid.d

```diff
@@ -193,6 +193,18 @@ public class Grid
 	}
 
 
+	public Cell cell(size_t row, size_t column)
```

Comment:
Here `size_t` doesn't refer to the variable itself but to the index. Indices in D are `size_t`

Reply:
Yes, but since you've initialized an arrays with size `int`, you're only able to index `int` values. So `size_t` might be needlessly bigger: it might even be used to index a non-existing value. I don't know how D handles this, but it wouldn't surprise me if it is undefined behavior.

Can you fact-check my assumptions @ljmf00?

Reply:
Not true all indices are declared as `size_t`. You can see an example here: https://dlang.org/spec/arrays.html#array-operations

Reply:
You can access the array with an `int` as the index value because the compiler might do an implicit cast, however, if you need to use the `length` property, for example to run through the array, you need to use `size_t`. An example of these can be seen here: https://dlang.org/spec/arrays.html#array-operations

Reply:
Yes, I understand that internally the language can implicitly cast to `size_t`, and use that anyways. But my problem is if there is a way to cause undefined behavior this way, by deliberately accessing an index that only fits in a `size_t` *(and that the array definitely doesn't have because in the initialization code it's defined as having max `int` values)*. As noted on https://dlang.org/spec/arrays.html#bounds:

> A program may not rely on array bounds checking happening, for example, ...

This sentence leads me to believe there's no guarantee the runtime will be always able to catch out-of-bounds conditions and throw an exception.